src/node/interfaces.cpp

```diff
@@ -883,11 +883,24 @@ class MinerImpl : public Mining
         return TestBlockValidity(state, chainman().GetParams(), chainman().ActiveChainstate(), block, tip, /*fCheckPOW=*/false, check_merkle_root);
     }
 
-    std::unique_ptr<CBlockTemplate> createNewBlock(const CScript& script_pub_key, bool use_mempool) override
+    std::unique_ptr<CBlockTemplate> createNewBlock(const CScript& script_pub_key, bool use_mempool,
+                                                   size_t coinbase_max_additional_weight,
+                                                   size_t coinbase_output_max_additional_sigops) override
     {
         BlockAssembler::Options options;
         ApplyArgsManOptions(gArgs, options);
 
+        Assume(coinbase_max_additional_weight <= DEFAULT_BLOCK_MAX_WEIGHT);
+        options.coinbase_max_additional_weight = coinbase_max_additional_weight;
+
+        Assume(options.coinbase_output_max_additional_sigops <= MAX_BLOCK_SIGOPS_COST);
```

Comment:
Why are we allowing for this large values?
shouldnt this be 
```suggestion
         Assume(coinbase_max_additional_weight <= MAX_STANDARD_TX_WEIGHT);
    options.coinbase_max_additional_weight = coinbase_max_additional_weight;
     Assume(options.coinbase_output_max_additional_sigops <= MAX_STANDARD_TX_SIGOPS_COST);
```

Reply:
`MAX_STANDARD_TX_WEIGHT` is a standardness rule. If we wanted to enforce it in the mining code, which would be unrelated to this PR, it shouldn't be through an assert / assume.